integrations/astrovault-adapter/src/dex.rs

```diff
@@ -36,59 +36,206 @@ pub const STABLE_POOL_FACTORY: &str =
 pub const RATIO_POOL_FACTORY: &str =
     "archway1zlc00gjw4ecan3tkk5g0lfd78gyfldh4hvkv2g8z5qnwlkz9vqmsdfvs7q";
 
+#[cfg(feature = "full_integration")]
+/// Copy of `query` but without deserialization as we don't know response type
+// fn query_without_deserialization(
+//     querier: &cosmwasm_std::QuerierWrapper,
+//     request: &cosmwasm_std::QueryRequest<cosmwasm_std::Empty>,
+// ) -> cosmwasm_std::StdResult<Binary> {
+//     let raw = cosmwasm_std::to_json_vec(request).map_err(|serialize_err| {
+//         StdError::generic_err(format!("Serializing QueryRequest: {serialize_err}"))
+//     })?;
+//     match querier.raw_query(&raw) {
+//         cosmwasm_std::SystemResult::Err(system_err) => Err(StdError::generic_err(format!(
+//             "Querier system error: {system_err}"
+//         ))),
+//         cosmwasm_std::SystemResult::Ok(cosmwasm_std::ContractResult::Err(contract_err)) => Err(
+//             StdError::generic_err(format!("Querier contract error: {contract_err}")),
+//         ),
+//         cosmwasm_std::SystemResult::Ok(cosmwasm_std::ContractResult::Ok(value)) => Ok(value),
+//     }
+// }
+
 #[cfg(feature = "full_integration")]
 impl Astrovault {
     fn fetch_pool_type(&self, deps: Deps, pool: &Addr) -> Result<PoolType, DexError> {
         // The pool type can be queried via identification of the factory contract associated with the pool
-        // We try the 3 different queries for one to match and get the factory address
-
-        let standard_config: Result<astrovault::standard_pool::query_msg::ConfigResponse, _> =
-            deps.querier.query_wasm_smart(
-                pool,
-                &astrovault::standard_pool::query_msg::QueryMsg::Config {},
-            );
-
-        if let Ok(config) = standard_config {
-            if config.factory.eq(STANDARD_POOL_FACTORY) {
-                return Ok(PoolType::ConstantProduct);
-            }
+        // We try the 3 different known factory addresses for one to match and get the factory address
+
+        #[cfg(feature = "full_integration")]
+        #[derive(cosmwasm_schema::serde::Deserialize)]
+        #[serde(rename_all = "snake_case", crate = "::cosmwasm_schema::serde")]
+        struct ConfigResponse {
```

Comment:
Can we put these types in a seperate mod?

Reply:
This will be the only type we will have, do we really need separate mod for struct with 1 type?

Reply:
moving it to separate mod in case of future use